src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
+
+    return response;
+  },
```

Comment:
PullProofAI

The `APIResponse` object seems to be a utility function for creating controller responses. It might be beneficial to encapsulate this functionality within a class or a module for better organization and maintainability.

```diff
+ export class APIResponse {
+   static create(
```

Overall, the code changes look good, but consider the suggestions provided for better TypeScript practices and code organization. 



<!-- This is an auto-generated comment by PullProofAI -->